File: src/main/java/payroll/Employee.java
package payroll;

import com.fasterxml.jackson.annotation.JsonIgnore;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.domain.Persistable;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Objects;
import java.util.UUID;

@Document(collection = "employees")
class Employee implements Persistable<UUID> {

    @Id
    private UUID id;
    private String name;
    private String role;

    public UUID getId() {
        return this.id;
    }

    @Transient
    @JsonIgnore
    @Override
    public boolean isNew() {
        return (getId() == null);
    }

    public String getName() {
        return this.name;
    }

    public String getRole() {
        return this.role;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setRole(String role) {
        this.role = role;
    }

    @Override
    public boolean equals(Object o) {

        if (this == o)
            return true;
        if (!(o instanceof Employee))
            return false;
        Employee employee = (Employee) o;
        return Objects.equals(this.id, employee.id) && Objects.equals(this.name, employee.name)
                && Objects.equals(this.role, employee.role);
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.id, this.name, this.role);
    }

    @Override
    public String toString() {
        return "Employee{" + "id=" + this.id + ", name='" + this.name + '\'' + ", role='" + this.role + '\'' + '}';
    }
}

File: kotlin/src/main/kotlin/payroll/Employee.kt
package payroll

import com.fasterxml.jackson.annotation.JsonIgnore
import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Transient
import org.springframework.data.domain.Persistable
import org.springframework.data.mongodb.core.mapping.Document
import java.util.*

@Document(collection = "employees")
class Employee : Persistable<UUID?> {
  @Id
  private var id: UUID? = null
  var name: String? = null
  var role: String? = null

  override fun getId(): UUID? {
    return id
  }

  @Transient
  @JsonIgnore
  override fun isNew(): Boolean {
    return getId() == null
  }

  fun setId(id: UUID?) {
    this.id = id
  }

  override fun toString(): String {
    return "Employee{id=$id, name='$name', role='$role'}"
  }

  override fun equals(other: Any?): Boolean {
    if (this === other) return true
    if (javaClass != other?.javaClass) return false

    other as Employee

    if (id != other.id) return false
    if (name != other.name) return false
    if (role != other.role) return false

    return true
  }

  override fun hashCode(): Int {
    var result = id?.hashCode() ?: 0
    result = 31 * result + (name?.hashCode() ?: 0)
    result = 31 * result + (role?.hashCode() ?: 0)
    return result
  }
}
